src/decoder/plugins/SidplayDecoderPlugin.cxx

```diff
@@ -65,6 +66,38 @@ static unsigned default_songlength;
 
 static bool filter_setting;
 
+#ifdef HAVE_SIDPLAYFP
+static const unsigned romSize = 8192;
+static std::unique_ptr<uint8_t[]> kernal(new uint8_t[romSize]());
+static std::unique_ptr<uint8_t[]> basic(new uint8_t[romSize]());
+
+static void loadRom(std::unique_ptr<uint8_t[]> &dump, const Path romPath)
+{
+	try
+	{
+		FileReader romDump(romPath);
+		if (romDump.GetSize() != romSize)
+		{
+			FormatWarning(sidplay_domain,
+				"Corrupted rom dump %s - %u bytes expected",
+				romPath.c_str(), romSize);
+		}
+		else
+		{
+			romDump.Read(dump.get(), romSize);
+			FormatInfo(sidplay_domain, "LOAD \"%s\",8,1",
+				romPath.c_str());
+		}
+	}
+	catch (...)
+	{
+		/* We break for nobody. Most tunes play without roms */
+		FormatWarning(sidplay_domain,
```

Comment:
If you catch all exceptions here, how does the caller know whether the buffers contain valid data? Maybe the best idea would be to return a `std::unique_ptr<uint8_t[]>`, and return `nullptr` on failure?
Or let the caller catch the exception? I think startup should fail if a ROM path was configured but failed to load. Even if some files play without a ROM - it's a misconfiguration.

Reply:
Agreed. MPD should fail if option is misconfigured.